kOS/pegas_util.ks

```diff
@@ -735,6 +743,56 @@ FUNCTION stageEventHandler {
 	setNextEvent(currentTime, eventDelay).
 }.
 
+//	Handles communication with other CPUs
+FUNCTION commsEventHandler {
+	LOCAL message IS CORE:MESSAGES:PEEK:CONTENT.	//	Get the oldest message in the queue but don't delete it yet
+
+	IF message["type"] = "request" {
+
+		//	Make sure data is in a list
+		LOCAL data IS message["data"].
+		IF NOT data:ISTYPE("List") {
+			SET data TO LIST(data).
+		}
+
+		//	List of available return data
+		LOCAL availableData IS LEXICON(
+			"liftOffTime", liftOffTime:SECONDS,
+			"launchAzimuth", mission["launchAzimuth"],
+			"upfgConverged", upfgConverged
+		).
+
+		//	Create a response to send back
+		LOCAL response IS LEXICON("type", "response").
+
+		//	Iterate through requested data and add to response
```

Comment:
Yeah, that makes sense. In case there is something wrong, should we return an actual error message (e.g. "Requested data not found") or just a simple "ERROR"?
Error message would be more descriptive for a human, but a simple "ERROR" would be easier to handle for a script.
Which one would you prefer or do you have any other idea?

Reply:
`"ERROR (Requested data not found)"` is equally easy to parse for the program that made the request (`RESPONSE:ENDSWITH("ERROR")` or something like that) and the human who might read the message on screen.

Reply:
Yeah, I forgot this is possible. Done now.